app/pages/project/instances/instance/tabs/SettingsTab.tsx

```diff
@@ -0,0 +1,166 @@
+          <SettingsGroup.Title>Auto-restart</SettingsGroup.Title>
+          <p>The auto-restart policy configured for this instance</p>
+        </SettingsGroup.Header>
+        <SettingsGroup.Body>
+          <ListboxField
+            control={form.control}
+            name="autoRestartPolicy"
+            label="Policy"
+            description="If unconfigured, this instance uses the default auto-restart policy, which may or may not allow it to be restarted."
+            placeholder="Default"
+            items={[
+              { value: '', label: 'Default' },
+              { value: 'never', label: 'Never' },
+              { value: 'best_effort', label: 'Best effort' },
+            ]}
+            required
+            className="max-w-none"
+          />
+          <FormMeta
+            label="Enabled"
+            helpText="If enabled this instance will automatically restart it if it enters the Failed state."
+          >
+            {instance.autoRestartEnabled ? 'True' : 'False'}{' '}
+            {instance.autoRestartEnabled && !instance.autoRestartPolicy && (
+              <span className="text-tertiary">(Project default)</span>
+            )}
+          </FormMeta>
+          <FormMeta
+            label="Cooldown expiration"
+            helpText="The time at which the auto-restart cooldown period for this instance completes. If n/a, then either the instance has never been automatically restarted, or the cooldown period has already expired"
+          >
+            {instance.autoRestartCooldownExpiration ? (
+              <>
+                {format(
+                  new Date(instance.autoRestartCooldownExpiration),
+                  'MMM d, yyyy HH:mm:ss zz'
+                )}{' '}
+                {new Date(instance.autoRestartCooldownExpiration) > now && (
+                  <span className="text-tertiary">
+                    ({formatDistanceToNow(new Date(instance.autoRestartCooldownExpiration))}
+                    )
+                  </span>
+                )}
+              </>
+            ) : (
+              <span className="text-tertiary">n/a</span>
+            )}
+          </FormMeta>
+          <FormMeta
+            label="Last auto-restarted"
+            helpText="When this instance was last automatically restarted. Empty if never auto-restarted."
+          >
+            {instance.timeLastAutoRestarted ? (
+              <>
+                {format(
+                  new Date(instance.timeLastAutoRestarted),
+                  'MMM d, yyyy HH:mm:ss zz'
+                )}
+              </>
+            ) : (
+              <span className="text-tertiary">n/a</span>
+            )}
+          </FormMeta>
+        </SettingsGroup.Body>
+        <SettingsGroup.Footer>
+          <div>
+            <LearnMore text="Auto-Restart" href={links.sshDocs} />
+          </div>
+          <Button size="sm" type="submit" disabled={!isDirty}>
+            Save
+          </Button>
```

Comment:
I think the relationship between the select and the table underneath could be clearer. The fact that the save button is at the bottom might be the problem. It makes it feel like the preview is inside the form and should therefore update live as a preview of what I'm saving. If the save button was next to the input and the state was below, it might be clearer that it would only update after I hit save.

Reply:
I think that's fair, though I'd prefer not to move the save button. This is the UX pattern for this settings form block universally, and it'll be a bit strange to do something different.

Perhaps I can try something like a treatment on the the enabled bit whilst the form has been changed. Cooldown expiration and auto-restart timestamp should not be affected by the policy change?

Reply:
![CleanShot 2025-02-03 at 11 03 14](https://github.com/user-attachments/assets/7a8ab535-bc1a-4bf4-b3e8-f9aad6c3e068)

Gave this a try. It might read as the policy is disabling the "enabled" field?

The best UX would be for us to infer what the field would be, but that might get a bit dicey. Especially if the behaviour changes in the future.

Reply:
Alternatively, we know based on the policy, it's either: true, false or project default. It's the inferred `autoRestartEnabled` that we're not sure of until the form is saved. So perhaps we hide the inferred value until it's submitted.

![CleanShot 2025-02-03 at 11 21 39](https://github.com/user-attachments/assets/409f7b15-cafd-4050-b6d0-8972a0ccae16)

Reply:
Neither of these solves the problem I ran into, which is that I didn't feel confident about whether/when my change is actually applied. Dimmed does look disabled. I don't think we have a pattern like this, but dotted line border or underline could make me think temporary/provisional. Another way would be to put the dropdown in a modal, but that feels very silly considering we have this perfect spot. What about just getting rid of the `enabled` part ? It's redundant with the listbox. Then all you need is a way of visually marking an unsaved change to the listbox. ...and I guess something to do with cooldown expiration while the listbox is changed to never but not saved. Kinda weird but figuring out a pattern here might be handy for inline editable table rows.